packages/webapp/src/_app/eos/secret-publisher.ts

```diff
@@ -61,25 +61,40 @@ export const encryptSecretForPublishing = async (
 
     const encryptedMessage = await encryptMessage(sessionKey, message);
 
-    console.info({
-        encryptedSessionKeys,
-        publisherKey,
-        recipientKeys,
-        transientKeyPair,
-        encryptedMessage,
-    });
+    const publisherKey = publicKeys[0];
+    const contractFormatEncryptedKeys: EncryptedKey[] = encryptedSessionKeys.map(
+        (encryptedKey, i) => ({
+            sender_key: publisherKey,
+            recipient_key: publicKeys[i],
+            key: encryptedKey,
+        })
+    );
 
-    // TODO: remove this test
-    console.info("testing decryption...");
-    await decryptPublishedMessage(
+    const decryptedMessage = await decryptPublishedMessage(
         encryptedMessage,
         publisherKey,
         transientKeyPair.publicKey.toString(),
         encryptedSessionKeys[0],
         info
     );
+    if (decryptedMessage !== message) {
+        throw new Error(
+            "an error occurred during encryption/decryption of the data"
+        );
+    }
```

Comment:
Decided to not remove this test? Or is this an oversight?

Reply:
No, I actually wanted to reinforce we are able to decrypt the message before returning it. The only thing to do is remove the logs for sensitive content.